DevelopingSpace/starchart
# #195 SSO and SLO additions

This Pull Request aims to close #195 

I used the previously added IDP and configured Starchart as a service provider that requests login and logout to it using redirect and post bindings. The SP metadata is generated by Samlify methods and then exposed on the /sp route.

The SAML response will have its assertions signed by the IDP.

The "POST-BINDING" for the logout actually is a get request and the response is found in the url.

## TO TEST THIS
You can take the code and pull it into a branch of your own or with gh cli or however you choose to do it and run the regular steps to run starchart. The login container should be up and running in docker and the login button will redirect to the IDP and the signout will also do the same. With the container running you can go here http://localhost:8081/simplesaml/saml2/idp/metadata.php?output=xhtml and see the IDP Metadata.

testing account for IDP:
username: user1
password: user1pass

You can also go into the network tab and check the flow of the SAML requests and responses through every HTTP request. The payload will have SAML response or SAML request and you can plug them into a [decoder](https://www.samltool.com/decode.php) and get the XML of the request or response. In this way you can see that the response for example has signed assertions.
